src/Components/Attempt/index.tsx

```diff
@@ -1,9 +1,11 @@
 import React, { useState } from "react";
+import CheckSolution from "../CheckSolution";
 import ColorContainer from "../ColorContainer";
 import HintsContainer from "../HintsContainer";
+import { DEFAULT_PEG_COLOR } from "../Utils/constants";
 import "./style.css"
 
-function Attempt(props: { isActive: boolean, action: Function, attemptColors: Array<string>, attemptHints: Array<string> }) {
+function Attempt(props: { isActive: boolean, action: Function, attemptColors: Array<string>, attemptHints: Array<string>, checkSolutionAction: Function }) {
```

Comment:
as discussed earlier, for more readability we can create a separate type here
and can also do it in the rest of the files as well